### PR TITLE
Recurring billing

### PR DESCRIPTION
**Pull request by NuckChorris**
Figured we should get code review going on this early.  Addresses #347 

Things which need to be done still:
 * [x] email templates

### CODE REVIEW

**Comment by NuckChorris** [comment]
I'd appreciate as many sets of eyes on this as possible, to make sure I didn't do anything stupid like accidentally bill users multiple times.  Pay special attention to the workers!

There's no tests covering the workers; due to lack of framework for functional testing of workers and some other test refactoring involving Sidekiq, I haven't finished those tests yet.

/cc @vikhyat @cybrox @vevix

**Comment by cybrox** [comment]
Looks good to my limited knowledge of things.

**Comment by vevix** [comment]
Other than my line comment, everything else looks good! :+1:

File: app/services/pro_membership_manager.rb [hunk 1]
```diff
@@ -44,8 +44,23 @@ def cancel!
   # Renew the user's pro membership. Make sure the plan they are on is recurring
   # first. If we don't succeed in charging the user then send a dunning email
   # and let their membership expire.
-  def renew!
-    # TODO we don't actually need this until one month after we launch PRO.
+  def renew!(attempt_number: 0)
```
Comment (vikhyat) [inline app/services/pro_membership_manager.rb]: Should we cap the `attempt_number`? Since we don't have email templates beyond a certain point?

**Comment by vikhyat** [comment]
:+1:

File: app/services/pro_membership_manager.rb (hunk 1 above)
Comment (NuckChorris) [inline app/services/pro_membership_manager.rb]: My assumption is that the number will be crammed into the template, not that it'll use a different template for each.

The attempts are already capped by the worker tbh

**Comment by NuckChorris** [comment]
Should be done.  Shall we merge and deploy?  (deployment I assume requires @vikhyat to restart the sidekiq workers?)

**Pull request merged**